src/storage/mod.rs

```diff
@@ -573,9 +545,10 @@ impl Storage {
         self.engine.clone()
     }
 
-    fn send(&self, cmd: Command, cb: StorageCb) -> Result<()> {
+    fn schedule(&self, cmd: Command, cb: StorageCb) -> Result<()> {
         fail_point!("storage_drop_message", |_| Ok(()));
-        box_try!(self.sendch.try_send(Msg::RawCmd { cmd: cmd, cb: cb }));
+        let worker = self.worker.lock().unwrap();
```

Comment:
Don't lock the worker and call `schedule` every time. You can add a worker scheduler as the Storage Field.